### PR TITLE
feat: improve code efficiency of `Invoke-VcfConfigReport`

### PR DESCRIPTION
**Pull request by GaryJBlake**
In order to have a good experience with our community, we recommend that you read the [contributing guidelines](https://github.com/vmware/powershell-module-for-cloud-foundation-reorting/blob/main/CONTRIBUTING_DCO.md) for making a pull request.

**Summary of Pull Request**

Improvements around code efficiency:
- Added $commandSwitch to create the -allDomains or -workloadDomain switch
- Refactored all `Publish-` calls to use $commandSwitch resulting in single command per element and reduction of ten lines of code per element
- 30% reduction in lines of code for `Invoke-VcfConfigReport`

Signed-off-by: Gary Blake <gblake@vmware.com>

**Type of Pull Request**

- [ ] This is a bug fix.
- [x] This is an enhancement or feature.
- [x] This is a code style / formatting update.
- [ ] This is a documentation update.
- [x] This is a refactoring update.
- [ ] This is a chore update
- [ ] This is something else.

**Related to Existing Issues**

Issue Number: N/A

**Test and Documentation Coverage**

- [x] Tests have been completed (for bug fixes / features).
- [ ] Documentation has been added / updated (for bug fixes / features).

**Breaking Changes?**

- [ ] Yes, there are breaking changes.
- [x] No, there are no breaking changes.

### CODE REVIEW

File: VMware.CloudFoundation.Reporting.psm1 [hunk 1]
```diff
@@ -536,64 +540,27 @@ Function Invoke-VcfConfigReport {
...
+                $vmOverridesHtml = Invoke-Expression "Publish-VmOverride -server $sddcManagerFqdn -user $sddcManagerUser -pass $sddcManagerPass $($commandSwitch)"; $reportData += $vmOverridesHtml
 
                 # Collecting Virtual Networking Using PowerShell Functions
                 Write-LogMessage -Type INFO -Message "Generating the Virtual Networking Configuration for $workflowMessage."
-                if ($PsBoundParameters.ContainsKey("allDomains")) {
-                    $vritualNetworkHtml = Publish-VirtualNetwork -server $sddcManagerFqdn -user $sddcManagerUser -pass $sddcManagerPass -allDomains
-                } else {
-                    $vritualNetworkHtml = Publish-VirtualNetwork -server $sddcManagerFqdn -user $sddcManagerUser -pass $sddcManagerPass -workloadDomain $workloadDomain
-                }
+                $vritualNetworkHtml = Invoke-Expression "Publish-VirtualNetwork -server $sddcManagerFqdn -user $sddcManagerUser -pass $sddcManagerPass $($commandSwitch)"; $reportData += $vritualNetworkHtml
```
Comment (tenthirtyam) [inline VMware.CloudFoundation.Reporting.psm1:559]: Minor, but we should search/replace `vritualNetworkHtml` with `virtualNetworkHtml`.

Comment (tenthirtyam) [reply VMware.CloudFoundation.Reporting.psm1:559]: But it doesn’t have impact as it’s consistent.

**Review by tenthirtyam — APPROVED** [review]
LGTM! 🚀

File: VMware.CloudFoundation.Reporting.psm1 (hunk 1 above)
Comment (GaryJBlake) [reply VMware.CloudFoundation.Reporting.psm1:559]: Easily fixed only two places on this line after the refactor

**Pull request merged**